packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -0,0 +1,142 @@
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { CfnDatabase } from './glue.generated';
+
+export interface IDatabase extends cdk.IConstruct {
+  /**
+   * The ARN of the catalog.
+   */
+  readonly catalogArn: string;
+
+  /**
+   * The catalog id of the database (usually, the AWS account id)
+   */
+  readonly catalogId: string;
+
+  /**
+   * The ARN of the database.
+   */
+  readonly databaseArn: string;
+
+  /**
+   * The name of the database.
+   */
+  readonly databaseName: string;
+
+  /**
+   * The location of the database (for example, an HDFS path).
+   */
+  readonly locationUri: string;
+
+  export(): DatabaseImportProps;
+}
+
+export interface DatabaseImportProps {
+  catalogArn: string;
+  catalogId: string;
+  databaseArn: string;
+  databaseName: string;
+  locationUri: string;
+}
+
+export interface DatabaseProps {
+  /**
+   * The name of the database.
+   */
+  databaseName: string;
+
+  /**
+   * The location of the database (for example, an HDFS path).
+   *
+   * @default a bucket is created and the database is stored under s3://<bucket-name>/<database-name>
+   */
+  locationUri?: string;
+}
+
+/**
+ * A Glue database.
+ */
+export class Database extends cdk.Construct {
+  /**
+   * Creates a Database construct that represents an external database.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param props A `DatabaseImportProps` object. Can be obtained from a call to `database.export()` or manually created.
+   */
+  public static import(scope: cdk.Construct, id: string, props: DatabaseImportProps): IDatabase {
+    return new ImportedDatabase(scope, id, props);
+  }
+
+  public readonly catalogArn: string;
+  public readonly catalogId: string;
+  public readonly databaseArn: string;
+  public readonly databaseName: string;
+  public readonly locationUri: string;
+
+  constructor(scope: cdk.Construct, id: string, props: DatabaseProps) {
+    super(scope, id);
+
+    if (props.locationUri) {
+      this.locationUri = props.locationUri;
+    } else {
+      const bucket = new s3.Bucket(this, 'Bucket');
+      this.locationUri = cdk.Fn.join('', ['s3://', bucket.bucketName, props.databaseName]);
+    }
+
+    this.catalogId = this.node.stack.accountId;
```

Comment:
This seems like something we may want to allow users to customize if they want, or does the catalog ID must be account ID

Reply:
It is but only AwsAccountId is supported at this time.